clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/AbstractKafkaConfigProperties.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.kafka;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Common configuration properties for Kafka clients.
+ */
+public abstract class AbstractKafkaConfigProperties {
+
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Common client config properties.
+     */
+    protected Map<String, String> commonClientConfig;
+    /**
+     * Client id to be applied if no {@code client.id} property is set in the common
+     * or specific client config properties.
+     */
+    protected String clientIdFallback;
```

Comment:
Instead of "fallback", I would rather call it a "default". Usually, this should not be set in config but in code. 
And: since each client needs an individual one (meaning each verticle) they are suffixed by a UUID (at least for producers, I'll add this for the consumer). So, maybe `clientIdDefaultPrefix` might be a better name. WDYT?

Reply:
Right, "default" is better. I would use `defaultClientIdPrefix` then.

Reply:
Using `prefix` here might be a bit confusing though, because it isn't used as a prefix here. The UUID suffix is applied at a different place.
How about changing this the getProducerConfig/getConsumerConfig/getAdminClientConfig methods to already include the UUID and an additional "name" parameter?

Reply:
I see the point with the term "prefix". You mean "name" as a third part of the effective client ID, like this is done in the following snipped? 
https://github.com/eclipse/hono/blob/18ba130bb51610c3f546a9d21621ef4a0e7aa1ad/clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/producer/AbstractKafkaBasedMessageSender.java#L246

And: when to generate the UUID? It must always be ensured that two clients cannot have the same client ID.

Reply:
Yes, that's the snippet I mean.

> And: when to generate the UUID?

Generate it when it is needed in `KafkaConsumerConfigProperties#getConsumerConfig()`.
Anyway, I think that change can be done in a separate PR.
As a preparation, I've already renamed the field/setter to `defaultClientIdPrefix` here.

Reply:
@b-abel I've made the aforementioned changes in PR https://github.com/eclipse/hono/pull/2441.